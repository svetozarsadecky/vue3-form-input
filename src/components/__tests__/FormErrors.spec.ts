import { describe, expect, it, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import MxFormState from '@/components/FormErrors.vue';

const factory = ({ props = {} } = {}) => mount(MxFormState, {
    attachTo: 'body',
    props: {
        ...props,
    },
    global: {
        directives: {
            tooltip: vi.fn()
        },
    }
});

describe('FormErrors', () => {
    it('should render component in success state', function () {
        const wrapper = factory();

        expect(wrapper.attributes().class).not.toContain('form-errors--is-error');
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render component in error state', function () {
        const wrapper = factory({
            props: {
                errors: ['error1', 'error2'],
            }
        });

        expect(wrapper.attributes().class).toContain('form-errors--is-error');
        expect(wrapper.html()).toMatchSnapshot();
    });
});
