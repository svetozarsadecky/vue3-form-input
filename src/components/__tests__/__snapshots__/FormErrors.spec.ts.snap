// Vitest Snapshot v1

exports[`FormErrors > should render component in error state 1`] = `
"<div class=\\"form-errors form-errors--is-error\\">
  <ul>
    <!--eslint-disable vue/no-v-html-->
    <li class=\\"form-errors__error\\">error1</li>
    <li class=\\"form-errors__error\\">error2</li>
  </ul>
</div>"
`;

exports[`FormErrors > should render component in success state 1`] = `
"<div class=\\"form-errors\\">
  <ul>
    <!--eslint-disable vue/no-v-html-->
  </ul>
</div>"
`;
